src/js/otus/sagas.js

```diff
@@ -19,59 +19,61 @@ import {
 } from "../app/actionTypes";
 import { apiCall, pushFindTerm, putGenericError } from "../utils/sagas";
 import * as otusAPI from "./api";
-
+import { createAction } from "@reduxjs/toolkit";
 const getCurrentOTUsPath = state => `/refs/${state.references.detail.id}/otus`;
+import { revertFailed, revertSucceeded } from "./actions";
 
 export function* updateAndGetOTU(apiMethod, action, actionType) {
     let response;
 
     try {
-        response = yield apiMethod(action);
+        response = yield apiMethod(action.payload);
     } catch (err) {
         yield putGenericError(actionType, err);
         response = err.response;
     }
 
-    const getResponse = yield otusAPI.get(action);
-    yield put({ type: actionType.SUCCEEDED, data: getResponse.body });
+    const getResponse = yield otusAPI.get(action.payload);
+    const curAction = createAction(actionType.SUCCEEDED);
+    yield put(curAction(getResponse.body));
```

Comment:
Stick to complete words for variable names: `currentAction`.

Also, it seems a bit heavy to use `createAction` here as it is used only to create the action object in the scope of this generator.

Reply:
My bad on the name, noted for the future. I don't disagree on createAction being heavy, and I had the same thought while doing it, but I was going for complete conversion to createAction. For now I have reverted to the old way using custom actions. We could still use createAction by defining success action for each in `actions.js` and then passing it to `updateAndGetOTU` but it is a lot less elegant than the current implementation. Let me know which way you want to go, happy with either.

Reply:
Let's keep the reverted form. Your suggestion is probably the most desirable solution, but it is not a high priority at the moment.